tools/release-plugin.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+BASE=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+. "$BASE/tools/includes/check-osx-bash-version.sh"
+. "$BASE/tools/includes/chalk-lite.sh"
+. "$BASE/tools/includes/proceed_p.sh"
+. "$BASE/tools/includes/plugin-functions.sh"
+. "$BASE/tools/includes/version-compare.sh"
+. "$BASE/tools/includes/normalize-version.sh"
+
+
+# Instructions
+function usage {
+	cat <<-EOH
+		usage: $0 [options] <plugin> <version>
+
+		Conduct a full release of a specified plugin through release branch creation. The <plugin> must be the slug of the plugin, such as plugins/jetpack
+
+		Options:
+			-a Release an alpha version
+			-b Release a beta version
+	EOH
+	exit 1
+}
+
+# Get the options passed and parse them.
+ARGS=('-p')
+ALPHABETA=
+while getopts "v:abh" opt; do
+	case ${opt} in
+		a)
+			ALPHABETA='-a'
+			ARGS+=("$ALPHABETA")
+		;;
+		b)
+			ALPHABETA='-b'
+			ARGS+=("$ALPHABETA")
+		;;
+		h)
+			usage
+		;;
+		?)
+			error "Invalid argument"
+			echo ""
+			usage
+		;;
+	esac
+done
+shift "$(($OPTIND -1))"
+
+# If there are more than two arguments, bail.
+[[ $# -gt 2 ]] && die "Too many arguments specified! Only provide a project and a version number. Got:$(printf ' "%s"' "$@")"$'\n'"(note all options must come before the project slug)"
+
+# Get the project slug.
+PROJECT=
+for ARG in "$@"
+	do
+		SLUG="${ARG#projects/}" # DWIM
+		SLUG="${SLUG%/}" # Sanitize
+		if [[ -e "$BASE/projects/$SLUG/composer.json" ]]; then
+			yellow "Project found: $ARG"
+			PROJECT="$SLUG"
+			shift
+			break
+		fi
+	done
```

Comment:
This will have weird behavior if someone does try to add extra non-option arguments before the slug. I'd lose the loop and process `$1`.

Reply:
Updated, let me know if that works better.